src/ReactResponsiveSelect.js

```diff
@@ -42,6 +43,15 @@ export default class ReactResponsiveSelect extends Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if(!isEqual(nextProps, this.props)) {
+      this.updateState({
+        type: actionTypes.UPDATE_VIA_PROPS,
+        value: { ...this.props, ...nextProps }
+      });
+    }
+  }
```

Comment:
This makes sense to me. We're checking the equivalence and dispatching a new event. What is actually left to do here?

Reply:
Answering my own question - the change event still fires and it shouldn't.

Reply:
Yes that is right. It stops the change event from re-firing.

Reply:
The change is still broadcasted using the latest commit. I am looking into not broadcasting the event with some logic in componentDidUpdate. I am about to test with my application code by removing the debounce logic and seeing if it works properly.

Reply:
I added this code and it allowed me to remove the debounce:

```javascript
  /* Broadcast change when there has been one */
  componentDidUpdate( prevProps, prevState ) {
    /* Update was done via props and should not be broadcasted */
    if (this.props.selectedValue === this.state.singleSelectSelectedOption.value) {
      return
    }

    // ...
  }
```

This only handles the singleSelect case though

Reply:
Nice one!

Dealing with Multiselect should be a case of adding a deep check against the multiSelect options
// Maybe something like 

if (
  this.props.selectedValue === this.state.singleSelectInitialIndex ||
  this.props.selectedValues === this.state.multiSelectInitialSelectedIndexes
) {
      return
    }
```